lib/puppet/pops/loader/typed_name.rb

```diff
@@ -24,7 +26,7 @@ def initialize(type, name)
     @name_parts = parts
 
     # Use a frozen compound key for the hash and comparison
-    @compound_name = "#{type}/#{name}".freeze
+    @compound_name = "#{name_authority}/#{type}/#{name}".freeze
```

Comment:
Does this have performance implications - all (in this impl) compound names will then start with the same string. Maybe we should consider reversing it?
Alternatively internalize it (make it a symbol) since comparison is then on identity.